core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -94,23 +93,25 @@ object Automation {
         } else {
             if (city.civ.gold < 0 && city.civ.stats.statsForNextTurn.gold <= 0)
                 yieldStats.gold *= 2 // We have a global problem
-            yieldStats.gold *= civPersonality.scaledFocus(PersonalityValue.Gold)
 
-            if (city.tiles.size < 12 || city.civ.wantsToFocusOn(Victory.Focus.Culture))
+            if (city.tiles.size < 12)
                 yieldStats.culture *= 2
-            yieldStats.culture *= civPersonality.scaledFocus(PersonalityValue.Culture)
 
-            if (city.civ.getHappiness() < 0 && !areWeRankingSpecialist) // since this doesn't get updated, may overshoot
+            if (city.civ.getHappiness() < 0)
                 yieldStats.happiness *= 2
-            yieldStats.happiness *= civPersonality.scaledFocus(PersonalityValue.Happiness)
-
-            if (city.civ.wantsToFocusOn(Victory.Focus.Science))
-                yieldStats.science *= 2
-            yieldStats.science *= civPersonality.scaledFocus(PersonalityValue.Science)
+        }
 
-            yieldStats.production *= civPersonality.scaledFocus(PersonalityValue.Production)
-            yieldStats.faith *= civPersonality.scaledFocus(PersonalityValue.Faith)
-            yieldStats.food *= civPersonality.scaledFocus(PersonalityValue.Food)
+        for (stat in Stat.values()) {
+            if (
+                stat == Stat.Culture && city.civ.wantsToFocusOn(Victory.Focus.Culture) ||
```

Comment:
There's a function to convert etat to victory type, no? We can make this concise

Reply:
No, I don't think there is. Also, there's not a focus for every stat (missing happiness and food) so such a  function would have to be in Civilization, not as a a function to convert like I created for personality types

Reply:
`wantsToFocusOn(focus: Victory.Focus)` does not accept null, I haven't checked if collections.contsins does either. Would make the call require an additional null check I would think (not at computer to look for a better solution). Maybe something like

```kotlin
        if (Victory.Focus.valueFor(stat) != null && civInfo.wantsToFocusOn(Victory.Focus.valueFor(stat)))
                buildingStats[stat] *= 2f
```
Not the cleanest thing in the world. The only other alternative is updating `wantsToFocusOn` to accept null correctly